### DISCUSSION TITLE
Describe the Backup Process

### DISCUSSION THREAD

**Discussion by Whytey** [turtle0x1/LxdMosaic#422] [Q&A]
Hi,

I have been playing with LxdMosaic within an lxd container.  The product seems to have matured since I last looked at it a few months ago, nice job!  

One of the things I was drawn by is the ability to do backups but there is very little info about these that I can find.

Could you describe the back up?  I traditionally do an export so I have a full backup I could import the backup as a recovery mechanism.  Where do the backed up files go?  Is it possible to automate this for new containers, also?

I would be happy to help you document what you tell me in a pull request, depending on the plan you have for documentation for the project.

Cheers,
Whytey

**Comment by turtle0x1** [comment]
Hey, thanks for your feedback - ill try my best to explain, a PR would be great.

Backups in LXDMosaic are a almost a direct copy of the `lxc export` command bound by all the same limitations (requires storage space to exist on the server)

How that works is you say to LXD "make a backup" which causes `LXD` to produce a `tar.gz` file that is then available to download LXDMosaic downloads that archive and tells LXD to delete the backup.

The backup files are stored under a path LXDMosaic can access except for when your using the SNAP which has restricted file system access.

The hierarchy used in this location is `$hostId/$project/$instance/$date-backup.tar.gz`

There is three important settings that should be set when backups are going to be used;

 - Timezone
 - Backup Directory
 - Backup History

Timezone: Regardless of where the end users web browser is, the backups will run according to the time zone set in the LXDMosaic setting (ignoring the system timezone)

Backup Directory: path to store the backup files, can only be changed outside of SNAP

Backup History: How long we keep the record of the backup (for nice graphs showing space used N months ago & to monitor growth / use)

Settings vaguely described [here](https://lxdmosaic.readthedocs.io/en/latest/LXDMosaic_Settings/)

Let me know if youd like more details / have more questions.
